src/datasets/arrow_dataset.py

```diff
@@ -2460,6 +2460,18 @@ def to_pandas(
                 for offset in range(0, len(self), batch_size)
             )
 
+    def add_column(self, column: dict):
```

Comment:
A dict may have several columns, this may be confusing.
I would expect `column` to be a `List` or a `np.ndarray`.
What do you think ?

Reply:
But we also need a "name" for the column. I was thinking of an input of the form: `{"column_name": column_data}`, where `column name` is a `str`, and  `column_data` is either a `List` or `np.ndarray`.

Reply:
@lhoestq, other option could be to pass 2 parameters: one for the column name and other for the column data.